Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut

```diff
@@ -961,16 +1000,123 @@ void function OnServerSelected( var button )
 
 	file.lastSelectedServer = serverIndex
 
+	// Count mods that have been successfully downloaded
+	int downloadedMods = 0;
+
 	// check mods
 	for ( int i = 0; i < NSGetServerRequiredModsCount( serverIndex ); i++ )
 	{
-		if ( !NSGetModNames().contains( NSGetServerRequiredModName( serverIndex, i ) ) )
+		string modName = NSGetServerRequiredModName( serverIndex, i );
+		string modVersion = NSGetServerRequiredModVersion( serverIndex, i );
+		print( format("Server requires mod \"%s\" v%s.", modName, modVersion) )
+
+		// If mod is locally installed already, no need to download it.
+		if (NSGetModNames().contains( modName ) && NSGetModVersionByModName( modName ) == modVersion && !NSIsModEnabled( modName ))
 		{
+			print( format("Required mod \"%s\" is locally there but disabled, enabling it.", modName) )
+			NSSetModEnabled( modName, true )
+			downloadedMods++
+			continue
+		}
+		
+		if ( !NSGetModNames().contains( modName ) )
+		{
+			string fullModName = "\"" + modName + "\""
+
+			// check if mod is verified
+			bool modIsVerified = false
+			if ( NSIsModVerified( modName, modVersion ) )
+			{
+				bool modIsVerified = true
+				string MB_UNIT = Localize( "#MEGABYTE_SMALL" )
+				NSDownloadMod( modName, modVersion )
+
+				// Downloading mod UI
+				DialogData dialogData
+				dialogData.header = Localize( "#DOWNLOADING_MOD_TITLE" )
+				dialogData.message = Localize( "#DOWNLOADING_MOD_TEXT", fullModName, modVersion )
+				dialogData.showSpinner = true;
+				// Prevent user from closing dialog
+				dialogData.forceChoice = true;
+
+				OpenDialog( dialogData )
+				// Save reference to UI elements, to update their content
+				var menu = GetMenu( "Dialog" )
+				var header = Hud_GetChild( menu, "DialogHeader" )
+				var body = GetSingleElementByClassname( menu, "DialogMessageClass" )
+
+				while( NSIsModBeingDownloaded(modName) )
+				{
+					// This array holds a bunch of progress information about mod.
+					ModDownloadProgress downloadStats = Build( NSGetCurrentDownloadProgress() );
+					bool isDownloading = downloadStats.progressType == eProgressType.DOWNLOAD;
+
+					// Mod is being downloaded.
+					if (isDownloading)
```

Comment:
If more downloading states might be added, this should be rewritten as a switch statement

Reply:
Do you see other verified mod installation steps than `DOWNLOADING` or `EXTRACTING`?
(I don't)